.github/workflows/golangci-lint.yml

```diff
@@ -0,0 +1,18 @@
+name: Go Linter
+on: push
```

Comment:
I would change this to run on PRs imo. Better to catch mistakes before they get pushed to main. I normally make an action called pr-checks which runs lint and also tests. That way ik that if I merge a pr at least it passes lint and the tests run. Also makes it so no one else can accidentally checkout my broken tests.